Microsoft.Toolkit/Diagnostics/Guard.Comparable.Numeric.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+
+#nullable enable
+
+namespace Microsoft.Toolkit.Diagnostics
+{
+    /// <summary>
+    /// Helper methods to verify conditions when running code.
+    /// </summary>
+    public static partial class Guard
+    {
+        /// <summary>
+        /// Asserts that the input value must be within a given distance from a specified value.
+        /// </summary>
+        /// <param name="value">The input <see cref="int"/> value to test.</param>
+        /// <param name="target">The target <see cref="int"/> value to test for.</param>
+        /// <param name="delta">The maximum distance to allow between <paramref name="value"/> and <paramref name="target"/>.</param>
+        /// <param name="name">The name of the input parameter being tested.</param>
+        /// <exception cref="ArgumentException">Thrown if (<paramref name="value"/> - <paramref name="target"/>) > <paramref name="delta"/>.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void IsCloseTo(int value, int target, int delta, string name)
```

Comment:
Do we need one for `long` as well?

Reply:
Sure thing! 😄

Done in https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3131/commits/41fea84172629f566aa090034ce3036ec6151312 and https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3131/commits/132d240ded5d271e1e92a28c5647fefa0800b06d, and also tweaked the surface area a bit 👍